File: src/main/java/org/sindeaux/autoconsulta/repositories/ConsultaRepository.java
package org.sindeaux.autoconsulta.repositories;

import org.sindeaux.autoconsulta.entityes.Consulta;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;


public interface ConsultaRepository extends JpaRepository<Consulta, Long> {

    public List<Consulta> findAlByMedicoId(Long id);

    public List<Consulta> findAllByPacienteId(Long id);
}

File: src/main/kotlin/com/sindeaux/autoconsulta/repository/ConsultaRepository.kt
package com.sindeaux.autoconsulta.repository


import com.sindeaux.autoconsulta.entity.Consulta
import org.springframework.data.jpa.repository.JpaRepository

interface ConsultaRepository : JpaRepository<Consulta?, Long?> {

    fun findAllByMedicoId(medico: Long): List<Consulta>;

    fun findAllByPacienteId(idPaciente: Long): List<Consulta>;

    fun findAllByLocalId(idLocal: Long): List<Consulta>;
}